CONTRIBUTING.md

```diff
@@ -0,0 +1,223 @@
+# Contributing
+
+This repo is built and maintained by the Galoy team. We welcome and appreciate new contributions and encourage you to check out the repo and [join our community Slack](https://join.slack.com/t/galoymoney-workspace/shared_invite/zt-rvnhsdb5-72AZCD_jzw6_Q05aCs0SgA) to get started.
+
+To help get you started, we will explain a bit about how we've laid out the code here and some of the practices we use. Our layout details fall mostly into two categories:
+- [Working with Types](#working-with-types)
+- [Understanding our Architecture](#architecture)
+
+---
+
+## Working with Types
+This codebase is implemented using a well-typed approach where we rely on Typescript for catching & enforcing certain kinds of checks we would like to perform.
+
+This approach helps to strenghthen guarantees that bugs/issues don't get to runtime unhandled if our types are implemented properly.
+
+### Our Approach
+We define all our types in declaration files (`*.d.ts`) where typescript automatically adds any types to the context without needing to explicitly import. Declaration files have limitations on what can be imported & instantiated within them, so in the few cases where types derive from instances of things we would usually have to use special tricks to derive them.
+
+The majority of our types are defined in the `domain` layer (see Architecture section below). We generally create types for the following scenarios:
+
+#### Symbol types
+These are unique types that are alternatives to generically typing things as Typescript's primitive types (e.g. `string`, `number`, `boolean`). Doing this helps us to add context to primitive types that we pass around and it allows the type checked to distinguish between different kinds of primitives throughout the code.
+
+For example, an onchain address and a lightning network payment request are both strings, but they aren't interchangeable as a data type. Instead of using `string` type for these types we would define as follows using a "unique symbol" trick:
```

Comment:
```suggestion
For example, an onchain address and a lightning network payment request are both strings, but they aren't interchangeable as a data type. Instead of using `string` type for these types we would define as follows using a "unique symbol":
```